`TestShowKeyspaceGroupPrimary` failed

## Flaky Test

### Which jobs are failing
```
2023-07-11T02:48:26.8749960Z --- FAIL: TestShowKeyspaceGroupPrimary (27.68s)
2023-07-11T02:48:26.8750146Z     keyspace_group_test.go:566: 
2023-07-11T02:48:26.8750605Z         	Error Trace:	/home/runner/work/pd/pd/tests/pdctl/keyspace/keyspace_group_test.go:566
2023-07-11T02:48:26.8751264Z         	            				/opt/hostedtoolcache/go/1.20.1/x64/src/runtime/asm_amd64.s:1598
2023-07-11T02:48:26.8751508Z         	Error:      	Received unexpected error:
2023-07-11T02:48:26.8752123Z         	            	invalid character 'F' looking for beginning of value
2023-07-11T02:48:26.8752392Z         	Test:       	TestShowKeyspaceGroupPrimary
2023-07-11T02:48:26.8752545Z     testutil.go:63: 
2023-07-11T02:48:26.8752944Z         	Error Trace:	/home/runner/work/pd/pd/pkg/utils/testutil/testutil.go:63
2023-07-11T02:48:26.8753732Z         	            				/home/runner/work/pd/pd/tests/pdctl/keyspace/keyspace_group_test.go:560
2023-07-11T02:48:26.8753980Z         	Error:      	Condition never satisfied
2023-07-11T02:48:26.8754340Z         	Test:       	TestShowKeyspaceGroupPrimary
```

### CI link
https://github.com/tikv/pd/actions/runs/5515219290/jobs/10055314959?pr=6774

### Reason for failure (if possible)


### Anything else

found log in cse
```
2023-08-08T03:48:52.5397779Z --- FAIL: TestShowKeyspaceGroupPrimary (27.01s)
2023-08-08T03:48:52.5397975Z     keyspace_group_test.go:565: 
2023-08-08T03:48:52.5398591Z         	Error Trace:	/home/runner/work/pd-cse/pd-cse/tests/pdctl/keyspace/keyspace_group_test.go:565
2023-08-08T03:48:52.5399283Z         	            				/opt/hostedtoolcache/go/1.20.1/x64/src/runtime/asm_amd64.s:1598
2023-08-08T03:48:52.5399542Z         	Error:      	Received unexpected error:
2023-08-08T03:48:52.5400166Z         	            	invalid character 'F' looking for beginning of value
2023-08-08T03:48:52.5400448Z         	Test:       	TestShowKeyspaceGroupPrimary
2023-08-08T03:48:52.5401220Z         	Messages:   	output: Failed to get the keyspace groups information: [500] "[PD:etcd:ErrEtcdTxnConflict]etcd transaction failed, conflicted and rolled back"
2023-08-08T03:48:52.5401374Z     testutil.go:63: 
2023-08-08T03:48:52.5401966Z         	Error Trace:	/home/runner/work/pd-cse/pd-cse/pkg/utils/testutil/testutil.go:63
2023-08-08T03:48:52.5402970Z         	            				/home/runner/work/pd-cse/pd-cse/tests/pdctl/keyspace/keyspace_group_test.go:560
2023-08-08T03:48:52.5403222Z         	Error:      	Condition never satisfied
2023-08-08T03:48:52.5403505Z         	Test:       	TestShowKeyspaceGroupPrimary
```

--- a/tests/pdctl/keyspace/keyspace_group_test.go
+++ b/tests/pdctl/keyspace/keyspace_group_test.go
@@ -561,8 +561,13 @@ func TestShowKeyspaceGroupPrimary(t *testing.T) {
 		args := []string{"-u", pdAddr, "keyspace-group"}
 		output, err := pdctl.ExecuteCommand(cmd, append(args, "1")...)
 		re.NoError(err)
+		if strings.Contains(string(output), "Failed") {
+			// It may be failed when meets error, such as [PD:etcd:ErrEtcdTxnConflict]etcd transaction failed, conflicted and rolled back
+			fmt.Println(string(output))
+			return false
+		}
 		err = json.Unmarshal(output, &keyspaceGroup)
-		re.NoErrorf(err, "output: %s", string(output))
+		re.NoError(err)
 		return len(keyspaceGroup.Members) == 2
 	})
 	for _, member := range keyspaceGroup.Members {

--- a/tests/pdctl/keyspace/keyspace_group_test.go
+++ b/tests/pdctl/keyspace/keyspace_group_test.go
@@ -563,7 +563,7 @@ func TestShowKeyspaceGroupPrimary(t *testing.T) {
 		re.NoError(err)
 		if strings.Contains(string(output), "Failed") {
 			// It may be failed when meets error, such as [PD:etcd:ErrEtcdTxnConflict]etcd transaction failed, conflicted and rolled back
-			fmt.Println(string(output))
+			re.Contains(string(output), "ErrEtcdTxnConflict", "output: %s", string(output))
 			return false
 		}
 		err = json.Unmarshal(output, &keyspaceGroup)

